src/main/java/lotto/collections/WinningTicket.java

```diff
@@ -0,0 +1,30 @@
+package lotto.collections;
+
+import lotto.domain.LottoNumber;
+import lotto.domain.LottoTicket;
+import lotto.service.WinningService;
+
+public class WinningTicket {
+
+	private WinningNumbers winningNumbers;
+	private LottoNumber bonusBall;
+
+	public WinningTicket(WinningNumbers winningNumbers, LottoNumber bonusBall) {
+		this.winningNumbers = winningNumbers;
+		this.bonusBall = validateBonusBall(bonusBall);
+	}
+
+	private LottoNumber validateBonusBall(LottoNumber bonusBall) {
+		if (this.winningNumbers.getWinningNumbers()
+			.stream()
+			.anyMatch(number -> number.getNumber().equals(bonusBall.getNumber()))) {
+			throw new RuntimeException("로또 번호와 보너스 볼은 중복 불가합니다.");
+		}
+		return bonusBall;
+	}
```

Comment:
해당 메서드 또한 객체 내부 상태를 get하지말고 WinningNumbers에게 validateBonusBall() 로직을 위임하여 메시지를 보내도록 한다면
```suggestion
	private WinningNumbers winningNumbers;
	private LottoNumber bonusBall;

	public WinningTicket(WinningNumbers winningNumbers, LottoNumber bonusBall) {
		this.winningNumbers = winningNumbers;
		this.bonusBall = winningNumbers.checkContainBonusBall(bonusBall);
	}
```
와 같이 객체에게 메시지를 보내 협력하도록 하고 캡슐화를 지킬 수 있을 것 같아요 :)
정답은 아니니 참고만 해주시면 감사하겠습니다 :)

Reply:
리뷰어님께서는 확실히 객체 간의 협력을 고민하시면서 설계하시네요.
저도 그러한 습관을 배워야겠습니다. 해당 피드백을 반영하도록 하겠습니다. 👍